addons/web_editor/static/src/js/wysiwyg/wysiwyg.js

```diff
@@ -1348,27 +1348,62 @@ const Wysiwyg = Widget.extend({
      * @param {Node} [params.htmlClass] Optionnal
      */
     openMediaDialog(params = {}) {
-        const sel = this.odooEditor.document.getSelection();
-
-        if (!sel.rangeCount) {
-            return;
-        }
-        const range = sel.getRangeAt(0);
         // We lose the current selection inside the content editable when we
         // click the media dialog button so we need to be able to restore the
         // selection when the modal is closed.
         const restoreSelection = preserveCursor(this.odooEditor.document);
 
-        const $editable = $(OdooEditorLib.closestElement(range.startContainer, '.o_editable') || this.odooEditor.editable);
-        const model = $editable.data('oe-model');
-        const field = $editable.data('oe-field');
-        const type = $editable.data('oe-type');
+        /**
+         * Looks up the attachment domain of the given element by traversing up the DOM tree
+         * until it finds an element with the necessary data attributes. If no such element is found,
+         * it returns the attachment domain of the editable element.
+         *
+         * @param {jQuery} $el - The jQuery element to start the search from.
+         * @returns {Object} An object containing the attachment domain information.
+         * @returns {string} resModel - The model name of the attachment.
+         * @returns {number|false} resId - The ID of the attachment, or false if it doesn't exist.
+         * @returns {Array} domain - The domain of the attachment.
+         * @returns {string|false} type - The type of the attachment, or false if it doesn't exist.
+         * @returns {string|false} field - The field of the attachment, or false if it doesn't exist.
+         */
+        function lookupAttachementDomain(el, top)  {
+            function extractAttachementDomain($el){
+                return {
+                    resModel: $el.data('oe-model') || 'ir.attachment',
+                    resId: $el.data('oe-id') || false,
+                    domain: $el.data('oe-media-domain') || [],
+                    type: $el.data('oe-type') || false,
+                    field : $el.data('oe-field') || false,
+                };
+            }
+
+            el = el.closest('.o_editable');
+
+            while (el) {
+                const $el = $(el);
+                if (($el.data('oe-model') && $el.data('oe-id')) || 
+                    $el.data('oe-media-domain')) {
```

Comment:
Are those actual `data-` attributes ? Because if yes, you might be able to replace that loop by a native selector. Something like:
```js
const el = params.node || this.odooEditor.document.getSelection().anchorNode;
const closestBrandedEditable = closestElement(el, '.o_editable[data-oe-model][data-oe-id],.o_editable[data-oe-media-domain]');
const $editable = $(closestBrandedEditable || this.odooEditor.editable);
// Code below this point is unchanged from 16.0.
```

Reply:
I tried it but jquery data() is not part of the DOM 

> Using the data() method to update data does not affect attributes in the DOM. To set a data-* [attr](https://api.jquery.com/attr/)ibute value, use attr.
https://api.jquery.com/data/

Reply:
Then there is the special `:data(...)` selector that can be used with jQuery. I wanted to give it a try before suggesting it to you but I can't seem to find a case where I get those special jQuery data. The commit message speaks of `mass_mailing` but even then when I start from an element of the mail I never find those data values on the `o_editable` element, even when trying manually. What am I missing ?

Reply:
I had to use an extension [1] to see these data in the dom

https://chrome.google.com/webstore/detail/jquery-debugger/dbhhnnnpaeobfddmlalhnehgclcmjimi

Reply:
Oh, that's another problem that I need to look into. When you first create the mailing, no data is attached, but once you save it, they are there.

Reply:
Well... I don't seem to have them even after saving. On which elements are they supposed to be ? The `o_mail_wrapper_td` one which is the one with `o_editable` ? Or another one ? I tried a few but maybe I did not try the right one.

Reply:
In this case, there is only one o_editable with no data, so we look at root odooEditor.editable

Reply:
Maybe I wrongly assumed that o_editable can be nested?

Reply:
No no, you're right that those classes can be nested. However, I discussed with @rdeodoo during lunch time and we're confused because I thought it was the double click on a image that was getting the wrong domain and not the replace button ? Here you seem to change the behavior when the user clicks on the replace button.

Reply:
No it's a change of behavior for both, Before we were only checking if we had a parent `o_editable` without making sure it had a domain before falling back on the `odooEditor.editable`. But now we make sure the `o_editable` has a domain. So the button was resolving to `odooEditor.editable` but it didn't check for `o_editable` so the behavior was incorrect and double-clicking was resolving to the `o_editable` which was incorrect too because it didn't have the domain.

Reply:
@dmo-odoo up

Reply:
Well I still see the whole manual loop with two levels of inner functions in the PR, and I still don't understand why the special jQuery selector `:data()` wouldn't work in this case.

Could you give the following code a try please ?

```js
const $el = $(params.node || this.odooEditor.document.getSelection().anchorNode);
const $closestBrandedEditable = $el.closest('.o_editable:data(oe-model):data(oe-id),.o_editable:data(oe-media-domain)');
const $editable = $closestBrandedEditable[0] || this.odooEditor.editable;
// Code below this point is unchanged from 16.0.
```

When jQuery is removed, those will need to use native `data-*` attributes and at that point my original suggested version will work too.

Reply:
This works, I pushed the change